### ISSUE TITLE
Confusing attach behaviors when multiple containers have the same image name

### ISSUE THREAD

**Issue by Chuxel** [microsoft/vscode-remote-release#1557]
Related to #1513

The new support in VS Code Insiders for remembering the default folder, extensions, when attaching has some behaviors I found counteractive when multiple containers with the same image existed on the system. Unclear if these are bugs, or confusing behaviors.  Two in particular:

1. The Containers Explorer has some odd behaviors
2. Configuration modified when attaching to one container is not applied to the other (and in fact is suggested to be reverted).

I can see multiple nodes with the same image on the same host (whether for deployment or just dev vs test) causing this situation pretty easily.

Explorer Repro:
1. Open a local terminal, `docker run -it alpine`
2. Open a second local terminal, `docker run -it alpine`
3. Start VS Code and go to the Container Explorer. You will see

    ![image](https://user-images.githubusercontent.com/312252/66003420-edd72400-e45a-11e9-84d4-15d95e3a8d67.png)

4. Attach to the first Alpine container in a new window
5. File > Open Folder... and select `/root`, install eslint in the container as well
6. Save `alpine.json`.  In the original window you will now see:

    ![image](https://user-images.githubusercontent.com/312252/66003584-68a03f00-e45b-11e9-9966-5e2db9bb1619.png)

7. Close the Alpine Container window
8. Now, attach to the second Alpine container  (that is still under Other Containers)
9. Note that you open `/root`, and anything else in `alpine.json` is applied, but the explorer still reflects the same state:

    ![image](https://user-images.githubusercontent.com/312252/66003584-68a03f00-e45b-11e9-9966-5e2db9bb1619.png)

This is counter intuitive - I would have expected this to either be tied to the container, or the image, but it's somewhere in-between.

Config Repro (continued from the explorer one):

10. Now, install another extension in this second container, and save `alpine.json`.
11. Close the window for the second container
12. Attach to the first container again
13. Note that:
    - The newly added extension in `alpine.json` is not added to the first container.
    - The `alpine.json` file actually appears again, but modified, with the other extension removed.

I would have expected that either both would get whatever was in `alpine.json`.

**Comment by chrmarti** [comment]
/cc @alexr00 for the second container not moving from Other Containers.

Extensions are only installed when attaching for the first time. After that the JSON only tracks the currently installed extensions.

I see that can be unexpected. It is consistent with the behavior with a devcontainer.json in a local project folder.

**Comment by Chuxel** [comment]
@chrmarti Yeah the fundamental challenge is that, unlike devcontainer.json, this file relates to multiple containers.  In this regard, it's not consistent.  If you change devcontainer.json for a given container, it wouldn't affect devcontainer.json for a different container.  I think that's the disconnect I'm seeing. If the json was tied to the container name (and thus the container), the way it is working now would make sense.

**Comment by chrmarti** [comment]
I don't want it to become overly automated:
- Imagine two windows being attached to two different Alpine containers and one installing a remote extension, should the second window immediately install the same remote extension?
- Or: One Alpine container might be used for one task and the other Alpine container for some other task with the two tasks requiring different extensions. Having both containers trying to stay in-sync with their extensions might be confusing.

We could notify the user during startup when the config file's list of extensions has changed and offer to install the additional ones.

**Comment by chrmarti** [comment]
Same for settings. (https://github.com/microsoft/vscode-remote-release/issues/1589)

**Comment by teohhanhui** [comment]
I have 2 `distrobox` / `podman` containers created from the same image (`quay.io/fedora/fedora-toolbox:40-aarch64`), named `fedora-40` and `asahi-krun` respectively.

I have previously attached to the first container (`fedora-40`) and created a named config file for it using the `Dev Containers: Open Named Container Configuration File` command.

It seems when I attached to the second container (`asahi-krun`), when I used the `Dev Containers: Open Named Container Configuration File` command, it actually gave me an empty JSON file, so it's correctly telling the 2 containers apart. However, even after I've created a named config file for it and reattached, when I open the terminal, it ends up in the wrong container (`fedora-40`).

So there are some remaining inconsistencies which give a confusing result...
